ServerCore/Pages/Submissions/Index.cshtml.cs

```diff
@@ -233,38 +284,70 @@ private async Task SetupContext(int puzzleId)
             Puzzle = await _context.Puzzles.Where(
                 (p) => p.ID == puzzleId).FirstOrDefaultAsync();
 
-            PuzzleState = await (PuzzleStateHelper
-                .GetFullReadOnlyQuery(
-                    _context,
-                    Event,
-                    Puzzle,
-                    Team))
-                .FirstAsync();
-
-            SubmissionViews = await (from submission in _context.Submissions
-                                     join user in _context.PuzzleUsers on submission.Submitter equals user
-                                     join r in _context.Responses on submission.Response equals r into responses
-                                     from response in responses.DefaultIfEmpty()
-                                     where submission.Team == Team &&
-                                     submission.Puzzle == Puzzle
-                                     orderby submission.TimeSubmitted
-                                     select new SubmissionView()
-                                     {
-                                         Submission = submission,
-                                         Response = response,
-                                         SubmitterName = user.Name,
-                                         FreeformReponse = submission.FreeformResponse,
-                                         IsFreeform = Puzzle.IsFreeform
-                                     }).ToListAsync();
-
-            Submissions = new List<Submission>(SubmissionViews.Count);
+            IsPuzzleForSinglePlayer = Puzzle.IsForSinglePlayer;
+            List<SubmissionView> submissionViews = new List<SubmissionView>();
+            if (Puzzle.IsForSinglePlayer)
+            {
+                PuzzleState = await SinglePlayerPuzzleStateHelper.GetOrAddStateIfNotThere(_context,
+                    Event,
+                    Puzzle,
+                    LoggedInUser.ID);
+
+                SubmissionViews = await (from submission in _context.SinglePlayerPuzzleSubmissions
+                                         join user in _context.PuzzleUsers on submission.Submitter equals user
```

Comment:
Don't we just want `where submission.Submitter == LoggedInUser.ID`? No join needed if so...